### PR TITLE
Create Check Answers page without Change links

### PR DESCRIPTION
**Pull request by woodpigeon**
- Create the 'check answers' (review) page
- Display a minimal set of data (whatever we have saved in all
  stories up to this point)
- Do not implement the 'Change' links as these are addressed
  in a later story
- Check Your Answers form delegates #rows to a presenter
- Use a presenter to build tabular data

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# This project uses environment variables for configuration, loaded by the dotenv gem.
```
Comment (reggieb) [inline .env.example]: The is already a .env.example at spec/dummy/.env.example

File: app/presenters/flood_risk_engine/enrollment_presenter.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    end
+
+    def organisation_address
+      AddressPresenter.new(organisation_primary_address).to_single_line
+    end
+
+    def correspondence_contact_name
+      return unless correspondence_contact
+      title = correspondence_contact.full_name
+      title += " (#{correspondence_contact.position})" unless correspondence_contact.position.blank?
```
Comment (reggieb) [inline app/presenters/flood_risk_engine/enrollment_presenter.rb]: I think `<<` is preferred to `+=` for string catenation.

http://stackoverflow.com/questions/4684446/why-is-the-shovel-operator-preferred-over-plus-equals-when-building-a

File: app/presenters/flood_risk_engine/tabular_enrollment_detail_presenter.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      arr.push row_builder.registration_date_row
+      arr.push(*row_builder.exemptions_rows)
+      arr.push row_builder.grid_reference_row
+      arr.push row_builder.organisation_type_row
+      arr.push row_builder.organisation_name_row
+      arr.push row_builder.organisation_address_row
+      arr.push row_builder.correspondence_contact_name_row
+      arr.push row_builder.correspondence_contact_telephone_row
+      arr.push row_builder.correspondence_contact_email_row
+      arr
```
Comment (reggieb) [inline app/presenters/flood_risk_engine/tabular_enrollment_detail_presenter.rb]: I think this is a simpler way to define an array:

```
[
  row_builder.registration_date_row,
  row_builder.exemptions_rows,
  row_builder.grid_reference_row,
  row_builder.organisation_type_row,
  row_builder.organisation_name_row,
  row_builder.organisation_address_row,
  row_builder.correspondence_contact_name_row,
  row_builder.correspondence_contact_telephone_row,
  row_builder.correspondence_contact_email_row
].flatten!
```

File: app/services/flood_risk_engine/address_services/search_service_failure.rb [hunk 4]
```diff
@@ -1,24 +1,26 @@
...
 
         Rails.logger.error "Address lookup service failed: #{ex}"
-
         Rails.logger.debug("EA::AddressLookup Config was: #{EA::AddressLookup.config.inspect}") if Rails.env.test?
 
         SearchServiceFailure.new(ex)
       end
-    end
 
+      def self.trap_vcr_errors_due_to_missing_env_vars(ex)
```
Comment (reggieb) [inline app/services/flood_risk_engine/address_services/search_service_failure.rb]: :smell:

File: app/views/flood_risk_engine/enrollments/_back_link.html.erb [hunk 5]
```diff
@@ -1,6 +1,6 @@
 <%= link_to(
       t("global.back"),
       enrollment_step_path(form.enrollment, form.enrollment.previous_step),
-      class: "back-link"
+      class: "link-back"
```
Comment (reggieb) [inline app/views/flood_risk_engine/enrollments/_back_link.html.erb]: The `back-link` class is used to identify the back link and replace it's functionality with JavaScript history step back. If you want to change the name, you also need to change it in app/assets/javascripts/flood_risk_engine/application.js

Comment (woodpigeon) [inline app/views/flood_risk_engine/enrollments/_back_link.html.erb]: Sorry my bad - I had added also broken this out as a partial in my branch but used `link-back` which is one of Polly's classes and adds the underline to the link. Will add `back-link ` back in also as both are required.

File: config/locales/en.yml [hunk 6]
```diff
@@ -1,4 +1,7 @@
 en:
+  time:
+    formats:
+      short: "%d %b %Y"
```
Comment (reggieb) [inline config/locales/en.yml]: Do you also need to do this for Date objects? Can be a bit of a gotcha. See http://nicholshayes.co.uk/blog/?p=223

File: config/locales/en.yml [hunk 7]
```diff
@@ -16,3 +19,4 @@ en:
   global:
     back: Back
     continue: Continue
+    back_link: Back
```
Comment (reggieb) [inline config/locales/en.yml]: Is there a problem with `t("global.back")` ?

File: config/locales/flood_risk_engine/enrollments/steps/show/generic.yml [hunk 8]
```diff
@@ -0,0 +1,37 @@
+---
```
Comment (reggieb) [inline config/locales/flood_risk_engine/enrollments/steps/show/generic.yml]: The generic locale file was dropped in a previous PR - is there a reason why you are bringing it back - or is this an unwanted merge relic?

File: spec/factories/organisations.rb [hunk 9]
```diff
@@ -15,5 +14,11 @@
         org_type ot.to_s
       end
     end
+
```
Comment (reggieb) [inline spec/factories/organisations.rb]: Do we need to add this commented out code?

File: config/locales/flood_risk_engine/enrollments/steps/show/generic.yml (hunk 8 above)
Comment (woodpigeon) [inline config/locales/flood_risk_engine/enrollments/steps/show/generic.yml]: A merge relic by the look of it

File: spec/presenters/flood_risk_engine/address_presenter_spec.rb [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+      context "when all address parts are present" do
+        let(:address) {
+          Address.new(premises: "a",
+                      street_address: "b",
+                      locality: "c",
+                      city: "d",
+                      postcode: "e")
+        }
+        it "returns the address as a comma delimited string" do
+          expect(subject.to_single_line).to eq("a,b,c,d,e")
```
Comment (reggieb) [inline spec/presenters/flood_risk_engine/address_presenter_spec.rb]: No space between each one? I'd expect the output to be `"a, b, c, d, e"` or is there a requirement to be more compact.

`Horizon House,Deanery Rd,Bristol BS1`

or

`Horizon House, Deanery Rd, Bristol BS1`

I think the second is easier to read.

Comment (woodpigeon) [inline spec/presenters/flood_risk_engine/address_presenter_spec.rb]: Doh, good spot thx!

**Comment by reggieb** [comment]
:ship:

**Pull request merged**